fem/bilininteg.cpp

```diff
@@ -1845,19 +1844,20 @@ void VectorFEMassIntegrator::AssembleElementMatrix2(
        && trial_fe.GetRangeType() == FiniteElement::VECTOR)
    {
       // assume test_fe is scalar FE and trial_fe is vector FE
-      int dim  = test_fe.GetDim();
+      int dim = test_fe.GetDim();
+      int spaceDim = Trans.GetSpaceDim();
```

Comment:
I'm not sure this whole section makes sense when `dim != spaceDim` but I still see some uses of `dim` that need to be converted to `spaceDim`. For example, the number of components in the scalar space should be `spaceDim` and in particular the element matrix should be `spaceDim*test_dof x trial_dof`. Also, the loops from `0` to `dim-1` should go up to `spaceDim-1`.

One exception is the case of matrix coefficient -- then the scalar space must have as many components as the number of rows in the matrix coefficient.

Reply:
Yeah, Do you have a testing code for that? I have only noticed the issue with both trial and test being vector spaces. If you know what needs to be fixed, go ahead and push here

Reply:
Done.

Reply:
thanks